pkg/controller/node/nodecontroller.go

```diff
@@ -644,12 +654,26 @@ func (nc *NodeController) monitorNodeStatus() error {
 	return nil
 }
 
-func nodeExistsInCloudProvider(cloud cloudprovider.Interface, nodeName string) (bool, error) {
-	instances, ok := cloud.Instances()
-	if !ok {
-		return false, fmt.Errorf("%v", ErrCloudInstance)
+type nodeExistanceChecker struct {
+	lock           sync.Mutex
+	instances      cloudprovider.Instances
+	lastUpdateTime time.Time
+	refreshPeriod  time.Duration
+}
+
+func (c *nodeExistanceChecker) nodeExistsInCloudProvider(cloud cloudprovider.Interface, nodeName string) (bool, error) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	now := time.Now()
+	if c.instances == nil || now.After(c.lastUpdateTime.Add(c.refreshPeriod)) {
+		instances, ok := cloud.Instances()
+		if !ok {
+			return false, fmt.Errorf("%v", ErrCloudInstance)
+		}
+		c.instances = instances
+		c.lastUpdateTime = now
```

Comment:
Yeah - I skimmed the interface description and it wasn't how I understood that:/ Will send update in a second.